Packs/Mimecast/Integrations/MimecastEventCollector/MimecastEventCollector.py

```diff
@@ -0,0 +1,567 @@
+            decoded_response = resp_body.decode('utf8')
+            json_response = json.loads(decoded_response)
+
+            if fail_reason := json_response.get('fail', []):
+                return_error(f'There was an error with siem events call {fail_reason}')
+            demisto.info('No more logs available')
+            return []
+        # Process log file
+        elif content_type == 'application/octet-stream':
+            file_name = resp_headers['Content-Disposition'].split('=\"')
+            file_name = file_name[1][:-1]
+
+            # Save the logs
+            events = self.write_file(file_name, resp_body)
+            # Save mc-siem-token page token to check point directory
+            if events:
+                self.token = resp_headers['mc-siem-token']
+
+            # return true to continue loop
+            return events
+
+        else:
+            # Handle errors
+            demisto.info('Unexpected response from siem logs')
+            headers_list = []
+            for header in resp_headers:
+                headers_list.append(header)
+                return_error(f'headers of failed request for siem errors: {headers_list}')
+
+    def process_siem_events(self, siem_json_resp: dict) -> list:
+        """
+        Args:
+            - siem_json_resp (list): a list of the siem events after extracted as json format
+
+        Returns:
+            - list: A flattened list of all fields before under the data part of the resopnse with some
+                        additional fields.
+        """
+        siem_log_type = siem_json_resp.get('type')
+        data: list = siem_json_resp.get('data', [])
+        events = []
+        for event in data:
+            event['type'] = siem_log_type
+            event['xsiem_classifier'] = 'siem_log'
+            self.convert_field_to_xdm_type(event)
+            events.append(event)
+
+        return events
+
+    @staticmethod
+    def convert_field_to_xdm_type(event: dict):
+        """
+        Args:
+            event (dict) - a siem event dict
+
+        Returns:
+            (None) this method works on the response as reference if the event had one of the following fields
+                    'IP', 'SourceIP', 'Recipient' they will the specified filed will be wrapped with an array.
+        """
+        for key in ['IP', 'SourceIP', 'Recipient', 'Rcpt']:
+            if key in event.keys() and not isinstance(event[key], list):
+                event[key] = [event[key]]
+
+    def get_req_object_siem(self):
+        """
+        Returns all data needed for the siem http request.
+        """
+        req_obj = {
+            'headers': self.client.prepare_headers(self.uri),   # type: ignore
+            'data': self.prepare_siem_log_data(),
+            'method': Method.POST,
+            'url': self.options.base_url + self.uri,
+            'verify': self.options.verify,
+        }
+        return req_obj
+
+    def prepare_siem_log_data(self):
```

Comment:
Suggesting a more suitable name.
```suggestion
    def prepare_siem_request_body(self):
```